ops/model.py

```diff
@@ -119,6 +122,41 @@
         'egress-subnets': List[str]
     })
 
+    _SecretMetadataDict = TypedDict('_SecretMetadataDict', {
+        'label': str,
+        'revision': str,
+        'expires': str,
+        'rotation': str,
+
+        # internally converted to SecretRotationPolicy enum
+        'rotates': Literal['never', 'hourly', 'daily', 'weekly',
+                           'monthly', 'quarterly', 'yearly'],
+    })
+
+    _SecretGetMetadataDict = Dict[str, _SecretMetadataDict]
+    _SecretGetContentsDict = Dict[str, Dict[str, str]]
+    _SecretGetResponseType = Union[str,
+                                   Dict[str, Union[
+                                       '_SecretGetContentsDict',
+                                       '_SecretGetMetadataDict']]]
+
+
+class SecretRotationPolicy(enum.Enum):
+    """Allowed secret rotation policies."""
+    never = 'never'  # the default in juju
```

Comment:
Two nits:

1) Elsewhere (in `pebble.py`) we have given these `ALL_UPPERCASE` names. This seems to be idiomatic in Python -- all the examples at https://docs.python.org/3/library/enum.html use this convention too. And a quick [public code search](https://sourcegraph.com/search?q=context:global+enum.Enum+lang:python&patternType=standard) shows other people follow this too. So I think we should use `NEVER`, `HOURLY`, and so on. At the very least to be internally consistent with `pebble.py`.
2) The class name `SecretRotationPolicy` is a bit of a mouthful. What about just `SecretRotate` (to match the `rotate` arg, and be more similar to the `SecretOwner` enum)?

Reply:
Thinking of that API, I used to have a colleague whose philosophy was: only expose primitive types in public signatures whenever possible.
`add_secret(rotate='never')` is to be preferred to `add_secret(rotate=SecretRotate.NEVER)`.
That said I'm not against passing enums either, I think we should allow both.
Personally I find it annoying that I have to import an enum when I know the string value that the hook tool is supposed to accept. 
But again, Unions are ugly too.
WDYT @benhoyt?

Reply:
I hear you; I too am fond of primitive types. :-) However, for strings which have semantic meaning and a limited range of values, it's too easy to make a typo (eg: accidentally typing `none` instead of `never`). So in the case of enums, I much prefer this being caught easily at runtime (or in some cases, import time or type check time).

Reply:
yeah, fair enough.